cpp/oneapi/dal/algo/linear_regression/train_types.cpp

```diff
@@ -33,6 +34,11 @@ class detail::v1::train_input_impl : public base {
 template <typename Task>
 class detail::v1::train_result_impl : public base {
 public:
+    table intercept;
```

Comment:
I see that it is possible to get an inconsistent state of train_result_impl when storing intercept and coefficients tables and model as a separate fields of the result. As they might be modified independently from each other.
Do we need set_intercept() and set_coefficients() at all? They only can lead to an object inconsistency and are not needed for compatibility with sklearn.

Reply:
Yes. In some sense these tables are not necessarily required since they can be obtained by decomposition of `packed_coefficients` table.
The thing is:
1) `train_results` is a separate entity from a `model`. We do not care about this tables after them being set. The infer stage is fed only with `model` itself
2) It can be hard to split table that is stored inside the model. Our table API does not support any kind of table views
3) User algorithms should be more or less agnostic about the internal structure of the model. In example there are many algorithms like kd-tree or decision-forest when user do not have access to the guts of model at all.

Reply:
Discussed